conformance/tests/httproute-listener-hostname-matching.go

```diff
@@ -66,8 +66,9 @@ var HTTPRouteListenerHostnameMatching = suite.ConformanceTest{
 			Backend:   "infra-backend-v3",
 			Namespace: ns,
 		}, {
-			Request:    http.ExpectedRequest{Host: "too.many.prefixes.bar.com", Path: "/"},
-			StatusCode: 404,
+			Request:   http.ExpectedRequest{Host: "multiple.prefixes.bar.com", Path: "/"},
+			Backend:   "infra-backend-v3",
+			Namespace: ns,
```

Comment:
I might be reading this wrong: this test proves the change by making sure the `*` match works right? Are we missing the test for the assertion that the suffix without a prefix doesn't match?

e.g. we need these tests:

- `*.ocean.example.com` matches `whales.ocean.example.com` and `sharks.ocean.example.com`
- `*.ocean.example.com` does **not** match `ocean.example.com`

Reply:
Good call, added test case for that. On a related note - I'm a big fan of those example domains. Trying to keep the surface area of this test small, but if anyone wants to move our tests to ocean themed example.com/net/org domains, that would also be great.

Reply:
fits right in with our already nautical software themes :joy: